### PR TITLE
PHP7 / Ubuntu 16.04 & PHP-FPM Pool Listen User/Group

### PR DESCRIPTION
**Pull request by ThatGerber**
This pull request adds support for PHP7 by moving around the order of operations for setting attributes to take into account settings in the upcoming Ubuntu 16.04, moving a few system calls into variables from the LWRP, and updating package names for php in the upcoming Ubuntu LTS 16.04. It also fixes a few foodcritic warnings.

This PR also pulls changes from #155, with updates. It creates a few extra pool configuration variables, allowing the pool listener to differ from the owner, and group to differ from the user. However, attributes have been set to mimic old behavior by default.

### CODE REVIEW

File: providers/pear_channel.rb [hunk 1]
```diff
@@ -24,25 +24,29 @@
...
+
 def whyrun_supported?
   true
 end
 
 action :discover do
   unless exists?
     Chef::Log.info("Discovering pear channel #{@new_resource}")
-    execute "#{node['php']['pear']} channel-discover #{@new_resource.channel_name}" do
+    r = execute "#{node['php']['pear']} channel-discover #{@new_resource.channel_name}" do
```
Comment (tas50) [inline providers/pear_channel.rb]: You shouldn't need to do any of the updated_by_last action mess if you use inline resources.  The inline resource method handles bubbling up the notifications

Comment (ThatGerber) [inline providers/pear_channel.rb]: Good to know. I'll remove the updated by last action code.

**Comment by jamescorkran** [comment]
@ThatGerber 
Forgive me for necroing this, but I had a question.  I am currently attempting to implement this cookbook to install PHP7 on our Ubuntu 16.04 machines.  However, it seems that before chef makes its way down the attributes it begins to check for PHP5 directories and errors with this:
```
Recipe: php::ini
  * template[/etc/php5/apache2/php.ini] action create
    * Parent directory /etc/php5/apache2 does not exist.
    ================================================================================
    Error executing action `create` on resource 'template[/etc/php5/apache2/php.ini]'
    ================================================================================

    Chef::Exceptions::EnclosingDirectoryDoesNotExist
    ------------------------------------------------
    Parent directory /etc/php5/apache2 does not exist.

    Resource Declaration:
    ---------------------
    # In /var/chef/cache/cookbooks/php/recipes/ini.rb

     21: template "#{node['php']['conf_dir']}/php.ini" do
     22:   source node['php']['ini']['template']
     23:   cookbook node['php']['ini']['cookbook']
     24:   unless platform?('windows')
     25:     owner 'root'
     26:     group node['root_group']
     27:     mode '0644'
     28:   end
     29:   variables(directives: node['php']['directives'])```

My question is:  Do I need to have php5 installed in order to install php7?
At the top of the attributes/default.rb file it seems to default to php5, but further down you've a case in place to check for platform version.  It does not reach this due to the default set at the start.

Am I missing something here?

**Comment by daften** [comment]
@jamescorkran  Did you find an answer to your question here?

**Pull request merged**